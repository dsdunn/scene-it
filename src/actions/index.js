export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (bool = false) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
});

export const eventsFetchSuccess = (events) => ({
  type: 'EVENTS_FETCH_SUCCESS',
  events: events
});

export const locationFetchSuccess = (location) => ({
  type: 'LOCATION_FETCH_SUCCESS',
  location: location
});

export const selectEvent = (event) => ({
  type: "SELECT_EVENT",
  event: event
})

export const unselectEvent = () => ({
  type: "UNSELECT_EVENT"
})

