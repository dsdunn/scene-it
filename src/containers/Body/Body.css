.main-body {
  width: 100%;
  display: block;
  background: center url('../../images/bar.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  height: 110vh;
}
.main-body::before {
  content: '';
  opacity: 0.4;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  position:absolute;
  height:110%;
  background-color: #261b16;
}

.map-div {
  display: flex;
  align-items: center;
  margin: 5% 4%;
  padding: 2%;
  height: 70vh;
}

.event-list {
  height: 65vh;
  width: 50%;
  overflow: scroll;
}


