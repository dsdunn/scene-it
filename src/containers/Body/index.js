import React from 'react';
import { connect } from 'react-redux';
import { EventCard } from '../../components/EventCard';
import  Map  from '../MapDiv';
import { mapKey } from '../../apiKey';
import './Body.css';
import {selectEvent} from '../../actions';
import {Details} from '../../components/Details';

const Body = (props) => {

  const eventList = () => props.events.map(event => <EventCard event={event} selectEvent={props.selectEvent}/>)

  return (
    <div className="Body">
      <div className="map-div">
        <Map 
          isMarkerShown
          googleMapURL={`https://maps.googleapis.com/maps/api/js?key=${mapKey}&v=3.exp&libraries=geometry,drawing,places`}
          loadingElement={<div style={{ height: `100%` }} />}
          containerElement={<div style={{ height: `60vh`, width: '60%' }} />}
          mapElement={<div style={{ height: `100%` }} />}
        />
      <div className="event-list">
        {eventList()}
      </div>
      </div>
      <Details event={props.selectedEvent}/>
    </div>
  );
}

export const mapStateToProps = (state) => ({
  events: state.events,
  selectEvent: state.selectEvent
})

export const mapDispatchToProps = (dispatch) => ({
  selectEvent: (event) => dispatch(selectEvent(event))
})

export default connect(mapStateToProps, mapDispatchToProps)(Body);