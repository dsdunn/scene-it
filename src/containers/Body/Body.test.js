import React from 'react';
import { shallow } from 'enzyme';
import Body from './index';
import { mapStateToProps, mapDispatchToProps } from './index';
import { selectEvent } from '../../actions';

describe('Body', () => {
  const mockState = {
    events: ['one', 'two', 'three'],
    selectedEvent: {eventId: 1}
  }
  const mockProps = {
    store: {getState: jest.fn(()=>mockState), subscribe: jest.fn(), dispatch: jest.fn()}
  }

  it('matches the snapshot', () => {
    let wrapper = shallow(<Body {...mockProps}/>)
    expect(wrapper).toMatchSnapshot();
  })

  it.skip('creates an eventList', () => {
    let wrapper = shallow(<Body {...mockProps} />)

    const actual = wrapper.instance().eventList()

    expect(actual.length).toEqual(3);
  })

  it.skip('renders an EventCard for every event', () => {
    let wrapper = shallow(<Body {...mockProps}/>)
    expect(wrapper.find(EventCard).length).toEqual(3)
  })
})

describe('mapStateToProps', () => {
  it('should return an object with events and a selected event', () => {
    const mockState = {
      events: ['one', 'two', 'three'],
      randomKey: 'this shouldn\'t show up',
      selectedEvent: {eventId: 1}
    }
    const expected = {
      events: ['one', 'two', 'three'],
      selectedEvent: {eventId: 1}
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  })
})

describe('mapDispatchToProps', () => {
  it('calls dispatch with selectEvent action when selectEvent is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = selectEvent({eventId: 1});
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.selectEvent({eventId: 1});
    expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })
})

