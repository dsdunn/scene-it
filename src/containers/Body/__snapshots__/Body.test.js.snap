// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    events={
        Array [
            "one",
            "two",
            "three",
          ]
    }
    selectEvent={[Function]}
    selectedEvent={
        Object {
            "eventId": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(BodyForm)) />,
        <div
          className="map-div"
>
          <div
                    className="event-list"
          >
                    <Unknown
                              event="one"
                              label={0}
                              selectEvent={[Function]}
                              selectedEvent={
                                        Object {
                                                  "eventId": 1,
                                                }
                              }
                    />
                    <Unknown
                              event="two"
                              label={1}
                              selectEvent={[Function]}
                              selectedEvent={
                                        Object {
                                                  "eventId": 1,
                                                }
                              }
                    />
                    <Unknown
                              event="three"
                              label={2}
                              selectEvent={[Function]}
                              selectedEvent={
                                        Object {
                                                  "eventId": 1,
                                                }
                              }
                    />
          </div>
          <withScriptjs(withGoogleMap(Connect(Map)))
                    containerElement={
                              <div
                                        style={
                                                  Object {
                                                            "height": "60vh",
                                                            "margin": "2%",
                                                            "width": "50%",
                                                          }
                                        }
                              />
                    }
                    googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places"
                    isMarkerShown={true}
                    loadingElement={
                              <div
                                        style={
                                                  Object {
                                                            "height": "100%",
                                                          }
                                        }
                              />
                    }
                    mapElement={
                              <div
                                        style={
                                                  Object {
                                                            "height": "100%",
                                                          }
                                        }
                              />
                    }
          />
</div>,
        <Unknown
          selectedEvent={
                    Object {
                              "eventId": 1,
                            }
          }
/>,
      ],
      "className": "main-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="event-list"
>
              <Unknown
                            event="one"
                            label={0}
                            selectEvent={[Function]}
                            selectedEvent={
                                          Object {
                                                        "eventId": 1,
                                                      }
                            }
              />
              <Unknown
                            event="two"
                            label={1}
                            selectEvent={[Function]}
                            selectedEvent={
                                          Object {
                                                        "eventId": 1,
                                                      }
                            }
              />
              <Unknown
                            event="three"
                            label={2}
                            selectEvent={[Function]}
                            selectedEvent={
                                          Object {
                                                        "eventId": 1,
                                                      }
                            }
              />
</div>,
            <withScriptjs(withGoogleMap(Connect(Map)))
              containerElement={
                            <div
                                          style={
                                                        Object {
                                                                      "height": "60vh",
                                                                      "margin": "2%",
                                                                      "width": "50%",
                                                                    }
                                          }
                            />
              }
              googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places"
              isMarkerShown={true}
              loadingElement={
                            <div
                                          style={
                                                        Object {
                                                                      "height": "100%",
                                                                    }
                                          }
                            />
              }
              mapElement={
                            <div
                                          style={
                                                        Object {
                                                                      "height": "100%",
                                                                    }
                                          }
                            />
              }
/>,
          ],
          "className": "map-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  event="one"
                  label={0}
                  selectEvent={[Function]}
                  selectedEvent={
                                    Object {
                                                      "eventId": 1,
                                                    }
                  }
/>,
                <Unknown
                  event="two"
                  label={1}
                  selectEvent={[Function]}
                  selectedEvent={
                                    Object {
                                                      "eventId": 1,
                                                    }
                  }
/>,
                <Unknown
                  event="three"
                  label={2}
                  selectEvent={[Function]}
                  selectedEvent={
                                    Object {
                                                      "eventId": 1,
                                                    }
                  }
/>,
              ],
              "className": "event-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "200",
                "nodeType": "function",
                "props": Object {
                  "event": "one",
                  "label": 0,
                  "selectEvent": [Function],
                  "selectedEvent": Object {
                    "eventId": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "201",
                "nodeType": "function",
                "props": Object {
                  "event": "two",
                  "label": 1,
                  "selectEvent": [Function],
                  "selectedEvent": Object {
                    "eventId": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "202",
                "nodeType": "function",
                "props": Object {
                  "event": "three",
                  "label": 2,
                  "selectEvent": [Function],
                  "selectedEvent": Object {
                    "eventId": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "containerElement": <div
                style={
                                Object {
                                                "height": "60vh",
                                                "margin": "2%",
                                                "width": "50%",
                                              }
                }
/>,
              "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places",
              "isMarkerShown": true,
              "loadingElement": <div
                style={
                                Object {
                                                "height": "100%",
                                              }
                }
/>,
              "mapElement": <div
                style={
                                Object {
                                                "height": "100%",
                                              }
                }
/>,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "selectedEvent": Object {
            "eventId": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(BodyForm)) />,
          <div
            className="map-div"
>
            <div
                        className="event-list"
            >
                        <Unknown
                                    event="one"
                                    label={0}
                                    selectEvent={[Function]}
                                    selectedEvent={
                                                Object {
                                                            "eventId": 1,
                                                          }
                                    }
                        />
                        <Unknown
                                    event="two"
                                    label={1}
                                    selectEvent={[Function]}
                                    selectedEvent={
                                                Object {
                                                            "eventId": 1,
                                                          }
                                    }
                        />
                        <Unknown
                                    event="three"
                                    label={2}
                                    selectEvent={[Function]}
                                    selectedEvent={
                                                Object {
                                                            "eventId": 1,
                                                          }
                                    }
                        />
            </div>
            <withScriptjs(withGoogleMap(Connect(Map)))
                        containerElement={
                                    <div
                                                style={
                                                            Object {
                                                                        "height": "60vh",
                                                                        "margin": "2%",
                                                                        "width": "50%",
                                                                      }
                                                }
                                    />
                        }
                        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places"
                        isMarkerShown={true}
                        loadingElement={
                                    <div
                                                style={
                                                            Object {
                                                                        "height": "100%",
                                                                      }
                                                }
                                    />
                        }
                        mapElement={
                                    <div
                                                style={
                                                            Object {
                                                                        "height": "100%",
                                                                      }
                                                }
                                    />
                        }
            />
</div>,
          <Unknown
            selectedEvent={
                        Object {
                                    "eventId": 1,
                                  }
            }
/>,
        ],
        "className": "main-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="event-list"
>
                <Unknown
                                event="one"
                                label={0}
                                selectEvent={[Function]}
                                selectedEvent={
                                                Object {
                                                                "eventId": 1,
                                                              }
                                }
                />
                <Unknown
                                event="two"
                                label={1}
                                selectEvent={[Function]}
                                selectedEvent={
                                                Object {
                                                                "eventId": 1,
                                                              }
                                }
                />
                <Unknown
                                event="three"
                                label={2}
                                selectEvent={[Function]}
                                selectedEvent={
                                                Object {
                                                                "eventId": 1,
                                                              }
                                }
                />
</div>,
              <withScriptjs(withGoogleMap(Connect(Map)))
                containerElement={
                                <div
                                                style={
                                                                Object {
                                                                                "height": "60vh",
                                                                                "margin": "2%",
                                                                                "width": "50%",
                                                                              }
                                                }
                                />
                }
                googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places"
                isMarkerShown={true}
                loadingElement={
                                <div
                                                style={
                                                                Object {
                                                                                "height": "100%",
                                                                              }
                                                }
                                />
                }
                mapElement={
                                <div
                                                style={
                                                                Object {
                                                                                "height": "100%",
                                                                              }
                                                }
                                />
                }
/>,
            ],
            "className": "map-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    event="one"
                    label={0}
                    selectEvent={[Function]}
                    selectedEvent={
                                        Object {
                                                            "eventId": 1,
                                                          }
                    }
/>,
                  <Unknown
                    event="two"
                    label={1}
                    selectEvent={[Function]}
                    selectedEvent={
                                        Object {
                                                            "eventId": 1,
                                                          }
                    }
/>,
                  <Unknown
                    event="three"
                    label={2}
                    selectEvent={[Function]}
                    selectedEvent={
                                        Object {
                                                            "eventId": 1,
                                                          }
                    }
/>,
                ],
                "className": "event-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "200",
                  "nodeType": "function",
                  "props": Object {
                    "event": "one",
                    "label": 0,
                    "selectEvent": [Function],
                    "selectedEvent": Object {
                      "eventId": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "201",
                  "nodeType": "function",
                  "props": Object {
                    "event": "two",
                    "label": 1,
                    "selectEvent": [Function],
                    "selectedEvent": Object {
                      "eventId": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "202",
                  "nodeType": "function",
                  "props": Object {
                    "event": "three",
                    "label": 2,
                    "selectEvent": [Function],
                    "selectedEvent": Object {
                      "eventId": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "containerElement": <div
                  style={
                                    Object {
                                                      "height": "60vh",
                                                      "margin": "2%",
                                                      "width": "50%",
                                                    }
                  }
/>,
                "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDuYzzorAhh3LMrRkNZPWD4r2TFZdhIzDI&v=3.exp&libraries=geometry,drawing,places",
                "isMarkerShown": true,
                "loadingElement": <div
                  style={
                                    Object {
                                                      "height": "100%",
                                                    }
                  }
/>,
                "mapElement": <div
                  style={
                                    Object {
                                                      "height": "100%",
                                                    }
                  }
/>,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "selectedEvent": Object {
              "eventId": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
